Docs: add a Frequently Asked Questions page

sphuber opened the pull request:
Fixes #3965 

Currently, the new FAQ page mostly focuses on common problems that
users encounter and direct either towards the mailing list or create
issues for. This FAQ page should become the central place to redirect to
in case of these common problems.

Besides the FAQ page itself, there are some other minor changes:

 * Improve `update installation` instructions to include the necessity
   of running `reentry` scan, and to be even more clear that the daemon
   should absolutely be stopped before upgrading, after having finished
   all running processes first.
 * Include link to FAQ page in the README
 * Remove outdated section from the Troublehooting section, since it was
   about Postgres versions that are no longer supported.

sphuber commented:
Currently, the FAQ entry in the sidebar can be expanded where each individual question becomes an item, because I use section markup for the questions. I am personally not sold on this, but am not sure how to suppress this. If you agree that it is a bit messy to have those long titles in the side bar and you know how to get rid of them, please let me know.

sphuber commented:
@csadorf if we find a place in the scaffolding for the FAQ, we could already review this and merge it I think. This is a new addition that I created before the docs reorganization.

csadorf commented:
> @csadorf if we find a place in the scaffolding for the FAQ, we could already review this and merge it I think. This is a new addition that I created before the docs reorganization.

Some of the questions addressed here should be moved to the "Troubleshooting" section as part of the intro chapter. The remaining content could then be placed as a distinct top-level section right after the How-Tos with the assumption that users should focus on going through the tutorial and relevant How-Tos before it makes sense to them to turn to FAQs. We then replace the "Development" panel with a FAQ panel on the landing page to "advertise" this more prominently. Thoughts?

ltalirz commented:
> The remaining content could then be placed as a distinct top-level section

This is one possibility but I find it a bit unflexible as there can be frequently asked questions on any stage - installation, tutorial, managing data, ...

E.g. I also still have on my to-do list to convert a document into a Q&A format that we prepared for the delegation of the Japanese National Institute of Informatics, which would be aimed more at the type of questions a manager would ask in order to decide whether AiiDA is worth giving a try (I'll send it to you in private).

csadorf commented:
As decided during team meeting we will place the FAQ section as the last subsection within the HowTo chapter for now.

csadorf commented:
@sphuber Can you rebase this on docs-revamp?

sphuber commented:
@csadorf rebased and ready for review

csadorf commented:
@sphuber I am very sorry, but looks like another rebase may be necessary.

sphuber commented:
not to worry. Rebase coming right up

sphuber commented:
@csadorf batter up!

csadorf commented on docs/source/howto/faq.rst:
```diff
@@ -0,0 +1,63 @@
+
+When I run a process, it finishes without any problems, but if I submit it to the daemon, it excepts. What is happening?
+========================================================================================================================
+In 99 out of a 100 cases, this is because a piece of code that is being executed by the process is not importable by the daemon.
+To determine exactly what, first set the loglevel to ``DEBUG`` by executing the command ``verdi config logging.aiida_loglevel DEBUG``.
```
```suggestion
This is almost always caused by an import issue.
To determine exactly what might be going wrong, first set the loglevel to ``DEBUG`` by executing the command ``verdi config logging.aiida_loglevel DEBUG``.
```

csadorf commented on docs/source/howto/faq.rst:
```diff
@@ -0,0 +1,63 @@
+========================================================================================================================
+In 99 out of a 100 cases, this is because a piece of code that is being executed by the process is not importable by the daemon.
+To determine exactly what, first set the loglevel to ``DEBUG`` by executing the command ``verdi config logging.aiida_loglevel DEBUG``.
+Then restart the daemon with ``verdi daemon restart`` for the changes to tak effect.
```
```suggestion
Then restart the daemon with ``verdi daemon restart`` for the changes to take effect.
```

csadorf commented on docs/source/howto/faq.rst:
```diff
@@ -0,0 +1,63 @@
+In 99 out of a 100 cases, this is because a piece of code that is being executed by the process is not importable by the daemon.
+To determine exactly what, first set the loglevel to ``DEBUG`` by executing the command ``verdi config logging.aiida_loglevel DEBUG``.
+Then restart the daemon with ``verdi daemon restart`` for the changes to tak effect.
+Run the command ``verdi daemon logshow`` in a separate terminal to see the logging output of the daemon and then submit the problematic calculation or workflow once more.
```
```suggestion
Run the command ``verdi daemon logshow`` in a separate terminal to see the logging output of the daemon and then submit the problematic calculation or workflow again.
```

sphuber commented on docs/source/howto/faq.rst:
```diff
@@ -0,0 +1,63 @@
+After you have stopped the daemon, make sure to run ``reentry scan`` before you restart the daemon with ``verdi daemon start``.
+
+
+I get an MissingEntryPointError or MultipleEntryPoint exception, saying that a particular entry point cannot be found. How can I fix this?
```
Turns out it is possible to have links in headers

sphuber commented:
Thanks @csadorf . Addressed all the changes and added one additional change: for three of the "longer" commands, I used the appropriate code blocks. I have some of the more basic ones, like `verdi status` and `verdi daemon stop` as inline. Didn't feel it was necessary to have code blocks for those, but let me know if you disagree

csadorf commented on docs/source/howto/faq.rst:
```diff
@@ -0,0 +1,79 @@
+
+I have updated the code of a :py:class:`~aiida.engine.processes.workchains.WorkChain`, :py:class:`~aiida.engine.processes.calcjobs.CalcJob` or whatever other Python code, but my Python shell instance does not seem to pick up the changes?
+=============================================================================================================================================================================================================================================
+The ``verdi shell`` is its own Python interpreter and also not automatically detect code changes and reload them.
```
```suggestion
The ``verdi shell`` is its own Python interpreter and does not automatically detect code changes.
Restart the shell to reload the latest version of the code.
```

csadorf commented on docs/source/howto/faq.rst:
```diff
@@ -0,0 +1,79 @@
+
+.. code-block:: ipython
+
+    In [1]:: computer.set_minimum_job_poll_interval(NUMBER_OF_SECONDS)
```
```suggestion
    In [1]: computer.set_minimum_job_poll_interval(NUMBER_OF_SECONDS)
```

csadorf commented on docs/source/howto/installation.rst:
```diff
@@ -258,9 +258,12 @@ Here are a few general tips that might improve the AiiDA performance:
 ==========================
 
+Whenever updating your AiiDA installation, make sure you follow these instructions **very carefully**, even when merely upgrading the patch version!
+Failing to do so, may leave your installation in a broken state, or worse may even damage your data, potentially irreperably.
```
```suggestion
Failing to do so, may leave your installation in a broken state, or worse may even damage your data, potentially irreparably.
```

sphuber commented:
> Just a few minor fixes, but otherwise this is good to go.

@csadorf done

csadorf commented:
@ltalirz Please let me know if you would like to have another look at this, otherwise I'll merge, thx!
